iogt_content_migration/management/commands/load_v1_users.py

```diff
@@ -0,0 +1,387 @@
+from wagtail.contrib.forms.utils import get_field_clean_name
+
+from tqdm import tqdm
+
+from home.models import Article, SectionIndexPage
+from questionnaires.models import Survey, UserSubmission, SurveyIndexPage, Poll
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--host',
+            default='0.0.0.0',
+            help='IoGT V1 database host'
+        )
+        parser.add_argument(
+            '--port',
+            default='5432',
+            help='IoGT V1 database port'
+        )
+        parser.add_argument(
+            '--name',
+            default='postgres',
+            help='IoGT V1 database name'
+        )
+        parser.add_argument(
+            '--user',
+            default='postgres',
+            help='IoGT V1 database user'
+        )
+        parser.add_argument(
+            '--password',
+            default='',
+            help='IoGT V1 database password'
+        )
+
+        parser.add_argument(
+            '--skip-locales',
+            action='store_true',
+            help='Skip data of locales other than default language'
+        )
+
+        parser.add_argument(
+            '--delete-users',
+            action='store_true',
+            help='Delete existing Users and their associated data. Use carefully'
+        )
+
+    def handle(self, *args, **options):
+        self.db_connect(options)
+
+        self.registration_survey_mandatory_group_ids = self.request_registration_survey_mandatory_groups()
+
+        self.content_type_map = dict()
+        self.comments_map = dict()
+        self.articles_map = dict()
+        self.surveys_map = dict()
+        self.polls_map = dict()
+        self.users_map = dict()
+
+        self.delete_users = options.get('delete_users')
+
+        self.clear()
+        self.stdout.write('Existing site structure cleared')
+
+        self.migrate()
+
+    def clear(self):
+        if self.delete_users:
+            self.stdout.write('Deleted Existing Users')
+            get_user_model().objects.all().delete()
+
+    def db_connect(self, options):
+        connection_string = self.create_connection_string(options)
+        self.stdout.write(f'DB connection string created, string={connection_string}')
+        self.v1_conn = psycopg2.connect(connection_string)
+        self.stdout.write('Connected to v1 DB')
+
+    def __del__(self):
+        try:
+            self.v1_conn.close()
+            self.stdout.write('Closed connection to v1 DB')
+        except AttributeError:
+            pass
+
+    def get_query_results_count(self, sql):
+        sql = f'select count(*) from ({sql}) as count_table'
+        cur = self.db_query(sql).fetchone()
+        return cur['count']
+
+    def with_progress(self, sql, iterable, title):
+        self.stdout.write(title)
+        return tqdm(iterable, total=self.get_query_results_count(sql))
+
+    def request_registration_survey_mandatory_groups(self):
```

Comment:
How well does this offer itself to automation? Would it make sense to have a flag via which these groups could be provided?

Reply:
Fair point. Let me add that in as well